pandas-stubs/io/excel/_base.pyi

```diff
@@ -1,188 +1,247 @@
-import abc
+from types import TracebackType
 from typing import (
     Any,
     Callable,
+    Generator,
+    Hashable,
+    Iterable,
+    Literal,
     Sequence,
     overload,
 )
 
+from odf.opendocument import OpenDocument
+from openpyxl.workbook.workbook import Workbook
 from pandas.core.frame import DataFrame
+import pyxlsb.workbook
+from xlrd.book import Book
 
 from pandas._typing import (
     Dtype,
     FilePath,
     ReadBuffer,
-    Scalar,
+    StorageOptions,
+    WriteExcelBuffer,
 )
 
+from pandas.io.common import IOHandles
+
 @overload
 def read_excel(
-    filepath: FilePath | ReadBuffer[bytes] | bytes,
-    sheet_name: list[int | str] | None,
+    io: FilePath
+    | ReadBuffer[bytes]
+    | bytes
+    | ExcelFile
+    | Workbook
+    | Book
+    | OpenDocument
+    | pyxlsb.workbook.Workbook,
+    sheet_name: list[int | str],
+    *,
     header: int | Sequence[int] | None = ...,
     names: list[str] | None = ...,
     index_col: int | Sequence[int] | None = ...,
-    usecols: int | str | Sequence[int | str | Callable] | None = ...,
-    squeeze: bool = ...,
-    dtype: str | dict[str, Any] | Dtype = ...,
-    engine: str | None = ...,
-    converters: dict[int | str, Callable] | None = ...,
-    true_values: Sequence[Scalar] | None = ...,
-    false_values: Sequence[Scalar] | None = ...,
-    skiprows: Sequence[int] | int | Callable | None = ...,
+    usecols: str | Sequence[int] | Sequence[str] | Callable[[str], bool] | None = ...,
```

Comment:
`Sequence[str]` will match `str` so no need to include both.

Reply:
Fixed.